### ISSUE TITLE
Redirection of HEAD requests end up as GETs

### ISSUE THREAD

**Issue by sngsweekeat** [axios/axios#1726]
#### Summary
In my node application, I'm using axios to perform a HEAD request, to try and find out what a user-submitted URL link is pointing to (content-type, size, etc). However, I find that, when the server issues a redirection, axios follows the redirection but uses GET instead, so the eventual response includes the actual content at the URL, which is not what I want.

My usage is pretty basic:
E.g.
```
        axios.head(url).then((response) => {
            console.log(response);
            // Do stuff with the response
        });
```
In the response object that is logged, I can see from the `request` attribute that the eventual request sent to the server after following redirects, is a GET:
```
  ...
  request: 
   ClientRequest {
     domain: null,
     ...
     _header: 'GET /blah/blah HTTP/1.1\r\nAccept: application/json, text/plain, */*\r\n...',
```

#### Context

- axios version: *0.18.0*
- Environment: *Node 8.4.0, MacOS*

**Comment by alcalbg** [comment]
Reported here: https://github.com/axios/axios/issues/1158

**Comment by chinesedfan** [comment]
Fixed by #1758.
